packages/radio-group/src/Radio.tsx

```diff
@@ -1,76 +1,216 @@
+      border: 3px solid rgba(184, 196, 194, 0.3);
+    }
+
+    &:focus + ${styledDiv.selector}:before {
+      border: 3px solid #9dd0e7;
+    }
+
+    &:disabled + ${styledDiv.selector} {
+      border-color: ${uiColors.gray.light1};
+      background-color: ${uiColors.gray.light2};
+    }
+  `,
+
+  [Variant.Light]: css`
+    &:checked + ${styledDiv.selector} {
+      background-color: #43b1e5;
+      border-color: #43b1e5;
+
+      &:after {
+        transform: scale(1);
+      }
+    }
+
+    &:hover + ${styledDiv.selector}:before {
+      border: 3px solid rgba(255, 255, 255, 0.2);
+    }
+
+    &:focus + ${styledDiv.selector}:before {
+      border: 3px solid ${uiColors.blue.base};
+    }
+
+    &:disabled + ${styledDiv.selector} {
+      background-color: ${uiColors.gray.dark2};
+      border-color: rgba(255, 255, 255, 0.15);
+    }
+  `,
+};
+
+const disabledChecked = {
+  [Variant.Default]: css`
+    &:disabled + ${styledDiv.selector} {
+      background-color: ${uiColors.gray.light1};
+      border-color: ${uiColors.gray.light1};
+
+      &:after {
+        background-color: ${uiColors.gray.light2};
+        transform: scale(1);
+      }
+    }
+  `,
+
+  [Variant.Light]: css`
+    &:disabled + ${styledDiv.selector} {
+      border-color: ${uiColors.gray.dark2};
+
+      &:after {
+        background-color: rgba(255, 255, 255, 0.3);
+        transform: scale(1);
+      }
+    }
+  `,
+};
+
 const inputStyle = css`
-  margin-right: 0.5em;
-  cursor: pointer;
+  height: 0;
+  width: 0;
+  opacity: 0;
 
-  &:disabled {
+  &:disabled:hover + ${styledDiv.selector}:before {
+    border: none;
+  }
+
+  &:disabled + ${styledDiv.selector} {
     cursor: not-allowed;
+
+    :after {
+      box-shadow: none;
+    }
   }
 `;
 
-const disabledButtonVariant = {
-  default: css`
-    color: ${colors.gray[5]};
-    cursor: not-allowed;
+const divVariantStyle = {
+  [Variant.Default]: css`
+    border: 2px solid ${uiColors.gray.dark1};
+    background-color: rgba(255, 255, 255, 0.31);
   `,
 
-  light: css`
-    color: ${colors.gray[4]};
-    cursor: not-allowed;
+  [Variant.Light]: css`
+    border: 2px solid ${uiColors.white};
+    background-color: rgba(255, 255, 255, 0.2);
   `,
 };
 
-const textStyle = css`
-  line-height: 0.9em;
+const divStyle = css`
+  position: relative;
+  transition: background-color 0.15s ease-in-out;
+  background-color: white;
+  display: inline-flex;
+  align-items: center;
+  justify-content: center;
+  flex-shrink: 0;
+  border-radius: 100px;
+  margin-right: 8px;
+
+  &:before {
+    content: '';
+    position: absolute;
+    border-radius: 100px;
+    top: -5px;
+    bottom: -5px;
+    left: -5px;
+    right: -5px;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+  }
+
+  &:after {
+    content: '';
+    box-shadow: inset 0 -1px 0 0 #f1f1f1, 0 1px 0 0 rgba(0, 0, 0, 0.08),
+      0 1px 1px 0 rgba(6, 22, 33, 0.22);
+    transition: transform 0.3s cubic-bezier(0.175, 0.885, 0.32, 1.275),
```

Comment:
We should have a separate transition for small and default sizes so that we can exaggerate the transformations in the smaller variant.

Also, we should be transitioning the background color. Consider adding a brief delay (a few ms) to the transform transition to refine this further.